web/src/components/storage/iscsi/InitiatorSection.jsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+
+import { Section } from "~/components/core";
+import { InitiatorPresenter } from "~/components/storage/iscsi";
+import { useInstallerClient } from "~/context/installer";
+import { useCancellablePromise } from "~/utils";
+
+export default function InitiatorSection() {
+  const { storage: client } = useInstallerClient();
+  const { cancellablePromise } = useCancellablePromise();
+  const [initiator, setInitiator] = useState();
+
+  useEffect(() => {
+    const loadInitiator = async () => {
+      setInitiator(undefined);
+      const name = await cancellablePromise(client.iscsi.getInitiatorName());
+      const ibft = await cancellablePromise(client.iscsi.getInitiatorIbft());
+      setInitiator({ name, ibft, offloadCard: "" });
+    };
+
+    loadInitiator().catch(console.error);
+
+    return client.iscsi.onInitiatorChanged(loadInitiator);
+  }, [cancellablePromise, client.iscsi]);
+
+  return (
+    <Section title="Initiator" iconName="settings">
+      <InitiatorPresenter
+        initiator={initiator}
+        client={client}
```

Comment:
NP: I do not see the point of this props here. As far as I can see, the presenter receives it for passing down to the form. This reminds me the [props drilling issue](https://kentcdodds.com/blog/prop-drilling)

So, I would just grab the client directly in the form, like you did here. That's one of the reasons we have the `useInstallerClient`.

Reply:
I have improved the situation a bit. Now, the client is used by the sections in order to get the necessary data, and then they pass the data and the client to the presenters. The presenters still need a client to perform the dbus calls corresponding to each possible action in the table. In another iteratation we could avoid passing clients from sections to presenters. At least, now the presenter is not passed to the form components.